src/apps/tools/OswAppCalculator.cpp

```diff
@@ -0,0 +1,448 @@
+        hal->gfx()->setTextSize(4);
+    }else{
+        hal->gfx()->setTextSize(3);
+        hal->gfx()->setTextColor(ui->getForegroundColor());
+        hal->gfx()->print(op);
+        hal->gfx()->setTextSize(4);
+    }
+
+    hal->gfx()->setTextLeftAligned();
+    hal->gfx()->setTextCursor(19, 140);
+
+     if(iNum == 8){
+        hal->gfx()->setTextSize(3);
+        hal->gfx()->setTextColor(ui->getSuccessColor());
+        hal->gfx()->print(pNum2);
+        hal->gfx()->setTextSize(4);
+    }else{
+        hal->gfx()->setTextSize(3);
+        hal->gfx()->setTextColor(ui->getForegroundColor());
+        hal->gfx()->print(pNum2);
+        hal->gfx()->setTextSize(4);
+    }
+
+    if(iNum == 9){
+        hal->gfx()->setTextColor(ui->getSuccessColor());
+        hal->gfx()->print(num2[0]);
+    }else{
+        hal->gfx()->setTextColor(ui->getForegroundColor());
+        hal->gfx()->print(num2[0]);
+    }
+
+    if(iNum == 10){
+        hal->gfx()->setTextColor(ui->getSuccessColor());
+        hal->gfx()->print(num2[1]);
+    }else{
+        hal->gfx()->setTextColor(ui->getForegroundColor());
+        hal->gfx()->print(num2[1]);
+    }
+
+    if(iNum == 11){
+        hal->gfx()->setTextColor(ui->getSuccessColor());
+        hal->gfx()->print(num2[2]);
+    }else{
+        hal->gfx()->setTextColor(ui->getForegroundColor());
+        hal->gfx()->print(num2[2]);
+    }
+
+    if(iNum == 12){
+        hal->gfx()->setTextColor(ui->getSuccessColor());
+        hal->gfx()->print(num2[3]);
+    }else{
+        hal->gfx()->setTextColor(ui->getForegroundColor());
+        hal->gfx()->print(num2[3]);
+    }
+
+    if(iNum == 13){
+        hal->gfx()->setTextColor(ui->getSuccessColor());
+        hal->gfx()->print(".");
+        hal->gfx()->print(num2[4]);
+    }else{
+        hal->gfx()->setTextColor(ui->getForegroundColor());
+        hal->gfx()->print(".");
+        hal->gfx()->print(num2[4]);
+    }
+
+    if(iNum == 14){
+        hal->gfx()->setTextColor(ui->getSuccessColor());
+        hal->gfx()->print(num2[5]);
+    }else{
+        hal->gfx()->setTextColor(ui->getForegroundColor());
+        hal->gfx()->print(num2[5]);
+    }
+
+    hal->gfx()->fillFrame(0, 160, 240, 10, ui->getSuccessColor());
+
+    hal->gfx()->setTextCenterAligned();
+    hal->gfx()->setTextSize(3);
+    hal->gfx()->setTextCursor(120, 210);
+    hal->gfx()->setTextColor(ui->getSuccessColor());
+    hal->gfx()->print(resultS);
+
+    hal->gfx()->setTextMiddleAligned();
+    hal->gfx()->setTextColor(ui->getForegroundColor());
+    hal->gfx()->setTextCursor(214, 57);
+    hal->gfx()->print("+"); 
+    hal->gfx()->setTextCursor(214, 184);
+    hal->gfx()->print("-");
+    hal->gfx()->setTextCursor(31, 184);
+    hal->gfx()->print(">"); 
+}
+
+
+void OswAppCalculator::setPNum1(){
+    OswHal* hal = OswHal::getInstance();
+
+    if (hal->btnHasGoneUp(BUTTON_3)){
+        i++;
+    }
+    if (hal->btnHasGoneUp(BUTTON_2)){
+        i--;
+    }
+    if (i < 0){
```

Comment:
How about `min(max(x, y), z)`?

Reply:
can you give me an example what you mean?